Main.py

```diff
@@ -613,7 +620,22 @@ def copy_world(world):
 
     for player in range(1, world.players + 1):
         if world.logic[player] in ('nologic', 'hybridglitches'):
-            create_hybridmajor_connectors(ret, player)
+            create_hybridmajor_connections(ret, player)
+
+    for region in world.regions:
```

Comment:
I assume this whole block addition is a mistake? No need to recommit anything, but I do need to know if I need to disregard this block addition. I had already previously pulled in the change that filters out hmg_entrances.

Reply:
No, this is needed. If you don't add this block the playthrough fails with items not being reachable, because the connections are not in the world model. This is the ugly fix that I mentioned on discord. They have to be skipped earlier, because the connections rely both on the previously created entrances as well as the later created portals for dungeons.

Reply:
Most seeds generate, but there are specific scenario's that don't. This one for example, hard requires clipping from Thieves -> Desert, then navigating to an entrance only accessible via that route, it fails without that chunk of code. Note the small change here, I'm only connecting the hmg entrances, whereas before I was skipping them:

```yaml
meta:
  players: 1
settings:
  1:
    logic: hybridglitches
    mode: inverted
    shuffle: crossed
start_inventory:
  1:
    - Pegasus Boots
    - Progressive Sword
    - Hammer
    - Fire Rod
placements:
  1:
    Desert Palace - Boss: Moon Pearl
    Desert Palace - Prize: Green Pendant
    Sahasrahla: Magic Mirror
entrances:
  1:
    two-way:
      Skull Woods Final Section: Desert Palace Exit (West)
      Skull Woods Second Section Door (West): Desert Palace Exit (East)
      Thieves Town: Thieves Town Exit
      Hyrule Castle Entrance (East): Desert Palace Exit (South)
      Hyrule Castle Entrance (West): Desert Palace Exit (North)
    entrances:
      Agahnims Tower: Pyramid Fairy
```

Reply:
I see, but then why this this part repeated?
`        copied_region = ret.get_region(region.name, region.player)
        copied_region.is_light_world = region.is_light_world
        copied_region.is_dark_world = region.is_dark_world
        copied_region.dungeon = region.dungeon
        copied_region.locations = [copied_locations[(location.name, location.player)] for location in region.locations]
        for location in copied_region.locations:
            location.parent_region = `copied_region``
and
`        for exit in region.exits:
            if exit.connected_region:
                dest_region = ret.get_region(exit.connected_region.name, region.player)
                #orig_ent = world.get_entrance(exit.name, exit.player)
                src_exit = ret.get_entrance(exit.name, exit.player)
                if exit.name not in [e.name for e in dest_region.entrances if e.connected_region is not None]:
                    if exit.name in [e.name for e in dest_region.entrances]:
                        src_exit.connected_region = dest_region
                    else:
                        `src_exit.connect(dest_region)``

Reply:
It looks like it can be trimmed down to the following:

```python
    for region in world.regions:
        copied_region = ret.get_region(region.name, region.player)
        for entrance in region.entrances:
            if entrance.name not in hmg_entrances:
                continue
            ret.get_entrance(entrance.name, entrance.player).connect(copied_region)
```

But without that, the playthrough fails. Honestly, I have no idea why, because `create_hybridmajor_connections` should be doing exactly this...